This adds the option to display the x- and y-axis on the graph

sandermvanvliet opened the pull request:
While looking for a way to use Knockout in combination with the D3 library I came across your binding which works like a charm. However I want to display the x- and y-axis on my graph so I thought to add it to the binding.
Currently the rendering of the axes is still within the graph itself, I'll see if I can make it a bit more pretty.

sandermvanvliet committed:
Added binding option to show axes on the plot

sandermvanvliet committed:
Updated README with example to enable axes

sandermvanvliet committed:
Fixed issue where binding fails when no options are specified

sandermvanvliet committed:
Added example for the showAxes option

sandermvanvliet committed:
Fixed id of the new example

gustavnikolaj commented:
Thanks for reaching out @sandermvanvliet 

The example is missing a background-color on the div with the binding, it looks as if nothing but the axis are rendered.

It looks promising, but I'd like the axis styling and placement fixed before merging it in and releasing it :-)

sandermvanvliet committed:
Added padding to the options to render the axes correctly

sandermvanvliet committed:
Added extra rect to make styling easier when axes are enabled

sandermvanvliet commented:
@gustavnikolaj I'm currently working on the styling issue, how does this look?
![image](https://cloud.githubusercontent.com/assets/2393265/6503268/8658c950-c32b-11e4-89e8-7a9820394d12.png)

gustavnikolaj commented:
Looks good to me :-)

Den torsdag den 5. marts 2015 skrev Sander van Vliet <
notifications@github.com>:

> @gustavnikolaj https://github.com/gustavnikolaj I'm currently working
> on the styling issue, how does this look?
> [image: image]
> https://cloud.githubusercontent.com/assets/2393265/6503268/8658c950-c32b-11e4-89e8-7a9820394d12.png
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/gustavnikolaj/knockout-d3-line-graph/pull/1#issuecomment-77341469
> .

sandermvanvliet commented:
Cool, those last two commits contain the changes for that, including the example bit about the styling.

gustavnikolaj commented:
There's some code style issues. Take a look at the diff here for example:
https://github.com/sandermvanvliet/knockout-d3-line-graph/commit/ffbd2867f1dd5164c82f9ef74109d68471791261

I have updated master so that it includes jshint and jscs to help solve some of these problems, but for CSS and HTML, you are on your own - I don't have any tooling around that atm. But the most important part is that you are using tabs (looks like 2 space width tabs :-)). The rest of the project uses 4 spaces instead of that, and I'd like that to change too, for consistency.

I have looked at the rest of the code, and it seems sane enough. :-)

After my recent changes to master, the pull request cannot be merged cleanly. It will probably be easiest for you to do the following to update the pull request:

First add the upstream repository as a remote.

```
$ git remote add gustavnikolaj https://github.com/gustavnikolaj/knockout-d3-line-graph.git
```

Then checkout your changes on a new branch for backup - in case something goes wrong - and go back to master.

```
$ git checkout -b pullrequest
$ git checkout master
```

Then rebase your changes on top of the changes on master. I have tried and it seems that you only get one conflict in `lib/knockout-line-graph.js` where you need to just choose your new updated version, and add a space after the `function` key word.

```
$ git rebase gustavnikolaj/master
// Here you will get a warning that a conflict happened.
$ git status
// Will tell you that the conflict is in lib/knockout-line-graph.js. Solve that conflict before going on...
$ git add lib/knockout-line-graph.js
$ git rebase --continue 
```

After that dance you will have rebased your branch on top of my changes. Then you need to force push your master branch of your fork.

```
$ git push origin master -f
```

Then you can run `npm lint` and have it assist you fix the styling problems. And after you have pushed the updates, the PR should be ready :-)

sandermvanvliet committed:
Fixed code style issues

sandermvanvliet commented:
Took me a while figuring out why the rebase failed but it seemed I had to do a pull of your master first in order to get it to work.
Code style issues should have been fixed now, I'll go see if I can make VIM behave itself in the future.

gustavnikolaj commented:
Regarding vim, you can take a look at the tabstop etc settings in my simple version of my .vimrc: https://github.com/gustavnikolaj/dotfiles/blob/master/simpleVimRcForServers.vimrc#L9 - unless you mean only for this project, in which case I think we need to look into adding a modeline in the files.

There's still some tabs in the CSS and the HTML, but other than that, it looks good.

gustavnikolaj commented:
I'm noticing what seems to be a 1px offset on the screen shot. The last tick on your axis seems to be outside the green area. Is that just me? If I'm not seeing ghosts, I'd like it to be adjusted :-)

sandermvanvliet committed:
Replaced tabs with spaces

sandermvanvliet committed:
Fixed postioning off-by-one-pixel issue

sandermvanvliet commented:
I've fixed the tabs and also the 1px offset thing. It appears to have something to do with the group that I added to make the plot render inside the axes. 
Good spot btw!

gustavnikolaj commented:
Thanks a lot for all your good work. :-) Merged!

gustavnikolaj commented:
Released as version 1.0.0 and added you to the contributors list in package.json. Thanks again.

sandermvanvliet commented:
That's awesome!
